go-controller/pkg/networkmanager/network_controller_test.go

```diff
@@ -0,0 +1,240 @@
+				{
+					Type:   "Accepted",
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+	}
+	nonPodNetworkRA := ratypes.RouteAdvertisements{
+		ObjectMeta: v1.ObjectMeta{
+			Name: testRAName,
+		},
+		Spec: ratypes.RouteAdvertisementsSpec{
+			TargetVRF: testVRFName,
+		},
+		Status: ratypes.RouteAdvertisementsStatus{
+			Conditions: []v1.Condition{
+				{
+					Type:   "Accepted",
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+	}
+	podNetworkRANotAccepted := podNetworkRA
+	podNetworkRANotAccepted.Status = ratypes.RouteAdvertisementsStatus{}
+	podNetworkRARejected := *podNetworkRA.DeepCopy()
+	podNetworkRARejected.Status.Conditions[0].Status = v1.ConditionFalse
+	podNetworkRAOutdated := podNetworkRA
+	podNetworkRAOutdated.Generation = 1
+
+	testNode := corev1.Node{
+		ObjectMeta: v1.ObjectMeta{
+			Name: testNodeName,
+		},
+	}
+	testNodeOnZone := corev1.Node{
+		ObjectMeta: v1.ObjectMeta{
+			Name: testNodeOnZoneName,
+			Annotations: map[string]string{
+				util.OvnNodeZoneName: testZoneName,
+			},
+		},
+	}
+	otherNode := corev1.Node{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "otherNode",
+		},
+	}
+
+	tests := []struct {
+		name            string
+		network         *ovncnitypes.NetConf
+		ra              *ratypes.RouteAdvertisements
+		node            corev1.Node
+		expectNoNetwork bool
+		expected        map[string][]string
+	}{
+		{
+			name:    "reconciles VRF advertisements for selected node of default node network controller",
+			network: defaultNetwork,
+			ra:      &podNetworkRA,
+			node:    testNode,
+			expected: map[string][]string{
+				testNodeName: {testVRFName},
+			},
+		},
+		{
+			name:    "reconciles VRF advertisements for selected node in same zone as default OVN network controller",
+			network: primaryNetwork,
+			ra:      &podNetworkRA,
+			node:    testNodeOnZone,
+			expected: map[string][]string{
+				testNodeOnZoneName: {testVRFName},
+			},
+		},
+		{
+			name:    "ignores advertisements that are not for the pod network",
+			network: defaultNetwork,
+			ra:      &nonPodNetworkRA,
+			node:    testNode,
+		},
+		{
+			name:    "ignores advertisements that are not for applicable node",
+			network: defaultNetwork,
+			ra:      &podNetworkRA,
+			node:    otherNode,
+		},
+		{
+			name:    "ignores advertisements that are not accepted",
+			network: defaultNetwork,
+			ra:      &podNetworkRANotAccepted,
+			node:    testNode,
+		},
+		{
+			name:            "fails for advertisements that are rejected",
+			network:         primaryNetwork,
+			ra:              &podNetworkRARejected,
+			node:            testNode,
+			expectNoNetwork: true,
+		},
+		{
+			name:            "fails for advertisements that are old",
+			network:         primaryNetwork,
+			ra:              &podNetworkRAOutdated,
+			node:            testNode,
+			expectNoNetwork: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+
+			config.OVNKubernetesFeature.EnableMultiNetwork = true
```

Comment:
I understand NADs are prerequisite for enabling RAs. but is there a requirement for cluster only with default network, but to enable RAs ?

Reply:
Yes because we will handle the default network through a special NAD which will be processed by the NAD controller. We could work around it, but not sure is worth the effort.